core/Inst.hpp

```diff
@@ -32,6 +32,34 @@ namespace olympia
     class Inst {
     public:
 
+        class RenameData{
+            public:
+                using SourceReg = std::pair<uint32_t, bool>;
+                
+                void setDestination(uint32_t destination){
```

Comment:
Consider using these namings to help understand:
setDestPhysReg
setDestArchReg
setArchPhysRegs
setArchPhysRegs

Reply:
Since we're keeping all references in the PRF, since we initialized the ARF into the PRF at startup, I think the current naming of "original" destination and destination make more sense. We want to know at retirement what was the "original" or older destination before we defined a new mapping, because we replace the map_table_ mapping when we map a new destination. We then can retire the original destination if it doesn't have any references, or we retire it when the last reference to it is retired.